src/lib/VncScreen.tsx

```diff
@@ -187,6 +190,21 @@ const VncScreen: React.ForwardRefRenderFunction<VncScreenHandle, Props> = (props
         logger.info(`Desktop name is ${e.detail.name}`);
     };
 
+    const _onServerVerification = async (e: { detail: { type: string, publickey: Uint8Array } }) => {
+        const rfb = getRfb();
+        if (onServerVerification) {
+            onServerVerification(e, rfb ?? undefined);
+            return;
+        }
+
+        if (e.detail.type === 'RSA') {
+            const fingerprint = await window.crypto.subtle.digest("SHA-1", e.detail.publickey)
```

Comment:
Currently the `fingerprint` is only logged, it would be better if we could expose the fingerprint (if present) by forwarding the ref to the parent component, so that the fingerprint can be programmatically verified if needed.

The easiest way to do it in my opinion is to assign it to the `RFB` object (if it has a `rfb.fingerprint` field?) or just on the `VNCScreenHandle` object.

P.S. I'm not sure the `fingerprint` needs to be used by a client in any scenario, but I assume that's the purpose of it - to make sure the fingerprint is in accordance with the VNC server

Reply:
I added support for react 19, which has caused some conflicts in the PR. Sorry for the inconvenience, it would be great if you could resolve the merge conflicts.